USAGE_EXTENSIBILITY.md

```diff
@@ -0,0 +1,136 @@
+## Extending Marked
```

Comment:
@Feder1co5oave, @UziTech, and @styfle: Could use some help on the content here, I think. Purpose is to document and demonstrate overriding/extending the renderer, lexer, and parser...seems we are light on documentation there.